src/box/memtx_tree.cc

```diff
@@ -910,12 +915,16 @@ memtx_tree_index_gc_run(struct memtx_gc_task *task, bool *done)
 	memtx_tree_t<USE_HINT> *tree = &index->tree;
 	memtx_tree_iterator_t<USE_HINT> *itr = &index->gc_iterator;
 
+	const bool is_func = index->gc_is_func;
```

Comment:
Why do you need this flag? Why not check `index->def` here to see if the index is functional?

Reply:
Index definition is already freed at this moment. See `index_delete`.